src/app_service/service/token/service.nim

```diff
@@ -604,3 +609,30 @@ QtObject:
     )
     self.threadpool.start(arg)
     return
+
+  proc getTokenPreferences*(self: Service): JsonNode =
+    try:
+      let response = backend.getTokenPreferences()
+      if not response.error.isNil:
+        error "status-go error", procName="getTokenPreferences", errCode=response.error.code, errDesription=response.error.message
+        return
+      return response.result
+    except Exception as e:
+      error "error: ", procName="getTokenPreferences", errName=e.name, errDesription=e.msg
+
+  proc updateTokenPreferences*(self: Service, tokenPreferencesJson: string) =
+    var updated = false
+    try:
+      let preferencesJson = parseJson(tokenPreferencesJson)
+      var tokenPreferences: seq[TokenPreferences]
+      if preferencesJson.kind == JArray:
+        for preferences in preferencesJson:
+          add(tokenPreferences, fromJson(preferences, TokenPreferences))
+      let response = backend.updateTokenPreferences(tokenPreferences)
```

Comment:
@Cuteivist  I have a question why are we saving this under the settings db? or creating a new settings for this? would that be more inline with how we manage settings variables

Reply:
got it. I understand now so its saved along with the tokens. as its a setting per token.